### PR TITLE
More GitHub actions tweaks

### PR DESCRIPTION
**Pull request by rbradford**
- build: Switch GitHub action ARM64 builds to musl
- build: Disable "fail fast" on x86-64 GitHub action
- build: Add libc to matrix for x86-64 tests
- build: Add some timeouts to integration test workflow

### CODE REVIEW

**Comment by rbradford** [comment]
Although splitting musl/glibc does introduce some extra overhead - I think the the fine grained control of being able to restart the builds separately sounds good. If we can prebuild the cloud-hypervisor binaries and kernel we should be able to streamline the build times further.

**Review by likebreath — APPROVED** [review]

**Comment by likebreath** [comment]
> Although splitting musl/glibc does introduce some extra overhead - I think the the fine grained control of being able to restart the builds separately sounds good. 

I agree.

> If we can prebuild the cloud-hypervisor binaries and kernel we should be able to streamline the build times further.

IIRC, Kata reuses the binary artifacts generated from other GH actions in their integration test jobs. We can adopt the similar approach to reduce the build time.  

@fidencio Would you please confirm my statement above and share some pointers? Thanks.

File: .github/workflows/integration-x86-64.yaml [hunk 1]
```diff
@@ -30,12 +33,8 @@ jobs:
       - name: Load openvswitch module
         run: sudo modprobe openvswitch
       - name: Run integration tests
-        run: scripts/dev_cli.sh tests --integration
+        timeout-minutes: 20
+        run: scripts/dev_cli.sh tests --integration --libc ${{ matrix.libc }}
```
Comment (likebreath) [inline .github/workflows/integration-x86-64.yaml:37]: I just noticed that the step `Run unit tests` is always running with gnu, say it needs to be updated to `scripts/dev_cli.sh tests --unit --libc ${{ matrix.libc }}`

**Comment by rbradford** [comment]
> > Although splitting musl/glibc does introduce some extra overhead - I think the the fine grained control of being able to restart the builds separately sounds good.
> 
> I agree.
> 
> > If we can prebuild the cloud-hypervisor binaries and kernel we should be able to streamline the build times further.
> 
> IIRC, Kata reuses the binary artifacts generated from other GH actions in their integration test jobs. We can adopt the similar approach to reduce the build time.
> 
> @fidencio Would you please confirm my statement above and share some pointers? Thanks.

It's not super easy - we can have GH build the assets and store them - would need to be on this workflow as an earlier job (cross workflow dependencies only work on the default branch) and then we would need to teach dev_cli.sh and internal scripts where to find "prebuilt" binaries (need a 2x2 matrix of libc & dbus_api). Ultimately when running "cargo test" most of the project is recompiled but that is a *debug* build so much cheaper.

**Pull request merged**